src/packages/ratmama/ratmama_parser.py

```diff
@@ -232,21 +243,34 @@ async def handle_ratsignal(ctx: Context) -> None:
             platform = Platforms["XB"]
 
         elif "o2" in part.casefold():
-            code_red = "o2 ok" not in part.casefold()
+            code_red = "o2 ok" not in message.casefold()
 
         else:
             system = part
 
+    if system and "signal" in system.lower():
```

Comment:
[Question] What case is this meant to catch?

Reply:
I honestly don't recall what case this catches,  but it was a buggy edge case.